agent/grpc/resolver/resolver.go

```diff
@@ -34,12 +35,21 @@ type Config struct {
 	Authority string
 }
 
-func NewServerResolverBuilder(cfg Config) *ServerResolverBuilder {
+func NewServerResolverBuilder(cfg Config) (*ServerResolverBuilder, error) {
+	if cfg.Authority == "" {
```

Comment:
At some point of my adventure trying to get `agent.TestAgent_JoinWAN_viaMeshGateway` to pass (in an earlier draft where there were changes) I suspected grpc resolver global cross-talk and arrived at this solution which would prevent issues across any parallel test case execution (or multiple agents within a single test case) without forcing the user to do anything.

Reply:
It's doable, but it would require exposing a test-only setting on `config.RuntimeConfig` to be able to pass something down into `agent/setup.go` where `NewServerResolverBuilder` is actually ever called in the common case. 

I guess alternatively maybe i could make `agent/setup.go` generate an Authority from `<datacenter>.<nodeName>` which even in a multi-agent unit test would be sufficiently unique?

Reply:
I've adjusted this to not use UUIDs, but in the common case in `agent/setup.go` it just does `<datacenter>.<nodeID>` and uses that for the authority which is pretty cheap and doesn't require punching a test-only flag down through RuntimeCOnfig